lxd/endpoints/network.go

```diff
@@ -0,0 +1,154 @@
+package endpoints
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/logger"
+	log "gopkg.in/inconshreveable/log15.v2"
+)
+
+// NetworkPublicKey returns the public key of the TLS certificate used by the
+// network endpoint.
+func (e *Endpoints) NetworkPublicKey() []byte {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+
+	return e.cert.PublicKey()
+}
+
+// NetworkAddress returns the network addresss of the network endpoint, or an
+// empty string if there's no network endpoint
+func (e *Endpoints) NetworkAddress() string {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+
+	listener := e.listeners[network]
+	if listener == nil {
+		return ""
+	}
+	return listener.Addr().String()
+}
+
+// NetworkUpdateAddress updates the address for the network endpoint, shutting
+// it down and restarting it.
+func (e *Endpoints) NetworkUpdateAddress(address string) error {
+	address = util.CanonicalNetworkAddress(address)
+	if address == e.NetworkAddress() {
+		return nil
+	}
+
+	logger.Infof("Update network address")
+
+	// First try to see if we can listen to this new port at all, so we
+	// don't close the old one (if any) in case of errors.
+	var listener net.Listener
+	if address != "" {
+		var err error
+		for i := 0; i < 10; i++ { // Ten retries over a second seems reasonable.
+			listener, err = net.Listen("tcp", address)
+			if err == nil {
+				break
+			}
+			logger.Debugf(" - retry binding TCP socket (%v)", err)
```

Comment:
Not sure we need this logged in the first place and if we do, this is going to render weirdly. The " - " prefix is only suitable when we're rendering something as part of the socket list on startup, this function is called when the user changes the address after startup, so that prefix isn't suitable here.

Reply:
Ok, I dropped that log message entirely in 5cb893bc603a9ed9cce08513ddc1dc4105a74a7a.